idx/memory/memory.go

```diff
@@ -130,6 +140,30 @@ func (m *MemoryIdx) Update(entry idx.Archive) {
 	m.Unlock()
 }
 
+func (m *MemoryIdx) addTags(def *schema.MetricDefinition) {
+	tags, ok := m.Tags[def.OrgId]
+	if !ok {
+		tags = make(map[string]map[string][]string)
+		m.Tags[def.OrgId] = tags
+	}
+	for _, tag := range def.Tags {
+		tagSplits := strings.SplitN(tag, "=", 2)
+		if len(tagSplits) < 2 {
+			// invalid
+			continue
+		}
+
+		if _, ok = tags[tagSplits[0]]; !ok {
+			tags[tagSplits[0]] = make(map[string][]string)
+		}
+
+		tags[tagSplits[0]][tagSplits[1]] = append(tags[tagSplits[0]][tagSplits[1]], def.Name)
```

Comment:
Do we need to init `tags[tagSplits[0]][tagSplits[1]]` first? this code will be much more readable if you say `tagName := tagSplits[0]` `tagValue := tagSplits[1]`

Reply:
It shouldn't be necessary to init slices, only maps. Ok, i'll name them